plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdDataTypeCache.java

```diff
@@ -77,7 +77,7 @@ protected synchronized void loadObjects(DBRProgressMonitor monitor, GenericStruc
                                 continue;
                             }
                             int fieldLength = JDBCUtils.safeGetInt(dbResult, "RDB$FIELD_LENGTH");
-                            int fieldScale = JDBCUtils.safeGetInt(dbResult, "RDB$FIELD_SCALE");
+                            int fieldScale = Math.abs(JDBCUtils.safeGetInt(dbResult, "RDB$FIELD_SCALE")); // For some reason, FireBird returns the negative value in the scale field.
```

Comment:
This behavior is described in the linked documentation:

> The **negative** number that specifies the scale for DECIMAL and NUMERIC columns—the number of digits after the decimal point

So, since it's expected, maybe use a negation instead of `abs`-ing it?